File: app/src/main/java/kr/hs/emirim/ohm/MainActivity.java
//package kr.hs.emirim.ohm;
//
//import android.os.Bundle;
//import android.view.View;
//import android.view.Menu;
//import com.google.android.material.navigation.NavigationView;
//import androidx.navigation.NavController;
//import androidx.navigation.Navigation;
//import androidx.navigation.ui.AppBarConfiguration;
//import androidx.navigation.ui.NavigationUI;
//import androidx.drawerlayout.widget.DrawerLayout;
//import androidx.appcompat.app.AppCompatActivity;
//import androidx.appcompat.widget.Toolbar;
//
//public class MainActivity extends AppCompatActivity {
//
//    private AppBarConfiguration mAppBarConfiguration;
//
//    @Override
//    protected void onCreate(Bundle savedInstanceState) {
//        super.onCreate(savedInstanceState);
//        setContentView(R.layout.activity_main);
//        Toolbar toolbar = findViewById(R.id.toolbar);
//        setSupportActionBar(toolbar);
//        DrawerLayout drawer = findViewById(R.id.drawer_layout);
//        NavigationView navigationView = findViewById(R.id.nav_view);
//        NavController navController = Navigation.findNavController(this, R.id.my_recycler_View);
//        NavigationUI.setupActionBarWithNavController(this, navController, mAppBarConfiguration);
//        NavigationUI.setupWithNavController(navigationView, navController);
//    }
//
//    @Override
//    public boolean onSupportNavigateUp() {
//        NavController navController = Navigation.findNavController(this, R.id.my_recycler_View);
//        return NavigationUI.navigateUp(navController, mAppBarConfiguration)
//                || super.onSupportNavigateUp();
//    }
//}

File: app/src/main/java/kr/hs/emirim/ohm/MainActivity.kt
package kr.hs.emirim.ohm

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.util.Log
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.database.FirebaseDatabase
import kotlinx.android.synthetic.main.activity_splash.*

class MainActivity : AppCompatActivity() {
    private lateinit var auth: FirebaseAuth

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)
        Handler().postDelayed({imageView.visibility = View.VISIBLE},5000L)
        Log.d("start", "oncreate")
        val ref = FirebaseDatabase.getInstance().getReference("users")
        auth = FirebaseAuth.getInstance()
        Log.d("start", "get auth")
        val user = auth.currentUser
        Log.d("start", "get user")
        if(user != null){
            ref.child(auth.currentUser!!.uid).get().addOnSuccessListener {
                if(it.value != null) {
                    Log.d("start", "there is a user")
                    startActivity(Intent(this, HomeActivity::class.java))
                    Log.d("start", "move home")
                    finish()
                }else{
                    Log.d("start", "no user")
                    startActivity(Intent(this, ProfileInitActivity::class.java))
                    Log.d("start", "move profile init")
                    finish()
                }
            }
        }else{
            startActivity(Intent(this, LoginActivity::class.java))
            finish()
        }
    }

}